src/api.rs

```diff
@@ -103,7 +103,9 @@ impl Context {
         deblock: Default::default(),
       };
 
-      self.fi.frame_type = if self.fi.number % 30 == 0 {
+      self.fi.order_hint = (self.fi.number % 30) as u32;
+
+      self.fi.frame_type = if self.fi.order_hint == 0 {
```

Comment:
LGTM too. Seems a bit weird to derive frame_type from order_hint but I guess it doesn't really matter.

Reply:
I agree that from a logical/readability point of view it might be worth using an intermediary variable and setting both `order_hint` and `frame_type` based on its value.